usaspending_api/common/etl/etl_table.py

```diff
@@ -0,0 +1,94 @@
+from collections import OrderedDict
+from django.utils.functional import cached_property
+from psycopg2.sql import Identifier, Literal, SQL
+from typing import List, MutableMapping, Optional
+from usaspending_api.common.helpers import sql_helpers
+
+
+class ETLTable:
+    """
+    Represents a table in the database.  Can be a local permanent table, a local
+    temporary table, or a dblinked permanent table.  Really just abstracts away much of
+    the database introspection bits and encapsulates table properties to reduce function
+    call interfaces.
+    """
+
+    def __init__(self, table_name: str, schema_name: Optional[str] = None, dblink_name: Optional[str] = None) -> None:
+        """
+        Initialize the ETL table.
+
+            table_name  - The name of the table this object represents.
+            schema_name - The schema in which the table lives.  IMPORTANT: Do not provide a
+                          schema for temporary tables as they do not have a fixed schema.
+            dblink_name - If the table lives in a remote database that can be queried via a
+                          PostgreSQL dblink, provide its name here.
+
+        """
+        self.table_name = table_name
+        self.schema_name = schema_name
+        self.dblink_name = dblink_name
+
+        if schema_name is None:
+            self.qualified_table_name = SQL("{}").format(Identifier(table_name))
+        else:
+            self.qualified_table_name = SQL("{}.{}").format(Identifier(schema_name), Identifier(table_name))
+
+    @cached_property
+    def columns(self) -> List[str]:
+        """ Returns the list of columns names from the table. """
+        return list(self.data_types)
+
+    @cached_property
+    def data_types(self) -> MutableMapping[str, str]:
+        """ Returns a mapping of columns names to database data types. """
+        sql = """
+            select      attname as column_name, pg_catalog.format_type(atttypid, atttypmod) as data_type
+            from        pg_attribute
+            where       attnum > 0 and attisdropped is false and attrelid = '{qualified_table_name}'::regclass
+            order by    attnum
+        """
+        sql = SQL(sql).format(qualified_table_name=self.qualified_table_name)
+        if self.dblink_name is not None:
```

Comment:
use `self.is_dblink()` here?

Reply:
heh... these stupid things are the reason I created `is_dblink`.  Apparently I missed a couple.